weedcoco/tests/test_validation.py

```diff
@@ -276,3 +280,15 @@ def test_coco_compatible_good(func, coco):
 def test_coco_compatible_bad(func, bad_coco):
     with pytest.raises(JsonValidationError):
         func(bad_coco, schema="compatible-coco")
+
+
+def test_pycocotools_quirks():
+    weedcoco = copy.deepcopy(BBOX_ONLY_WEEDCOCO)
+    with pytest.raises(Exception, match="datasetType not supported"):
```

Comment:
This is an interesting test... While it justifies the need for a fix, it's highly dependent on implementation details in pycocotools! I'm okay with it for now, but expect this clause to be removed when we upgrade pycocotools :)